server/src/internalClusterTest/java/org/elasticsearch/health/HealthMetadataServiceIT.java

```diff
@@ -66,7 +66,12 @@ public void testEachMasterPublishesTheirThresholds() throws Exception {
                 var healthMetadata = HealthMetadata.getFromClusterState(internalCluster.clusterService().state());
                 var diskMetadata = healthMetadata.getDiskMetadata();
                 assertThat(diskMetadata.describeHighWatermark(), equalTo(watermarkByNode.get(electedMaster)));
-                assertThat(diskMetadata.highMaxHeadroom(), equalTo(maxHeadroomByNode.get(electedMaster)));
+                // The value of the setting `cluster.routing.allocation.disk.watermark.high.max_headroom` depends upon the existence of
+                // `cluster.routing.allocation.disk.watermark.high`. Check {@link CLUSTER_ROUTING_ALLOCATION_HIGH_DISK_MAX_HEADROOM_SETTING}
+                assertThat(
+                    diskMetadata.highMaxHeadroom(),
+                    equalTo(percentageMode ? maxHeadroomByNode.get(electedMaster) : ByteSizeValue.ofGb(150))
```

Comment:
It took me a while to figure out the relation between the `percentageMode` and the maxHeadroom. Maybe add a comment here that the headroom is not set in the settings if `percentageMode` is false. Or, if possible, set null in the hashmap that passes the value. This way while reading the test I can see how they relate.

What do you think?

Reply:
that's why I added the comment to each of these tests, to clearly state that the value depends on the value of other settings... craziness